lib/net.c

```diff
@@ -27,11 +27,7 @@
 #include "gatekeeper_net.h"
 #include "gatekeeper_config.h"
 
-/* 
- * Parameters will be detected automatically. 
- */
-static uint32_t num_ports;
-
+static struct net_config *net_conf;
```

Comment:
Pass this struct as parameters to the functions that need it. Please avoid unnecessary global variables.